client/a8c-for-agencies/sections/referrals/referrals-list/commissions-column.tsx

```diff
@@ -1,19 +1,72 @@
+import { Tooltip } from '@automattic/components';
 import { formatCurrency } from '@automattic/format-currency';
-import { useEffect, useState } from 'react';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState, useRef } from 'react';
 import TextPlaceholder from 'calypso/a8c-for-agencies/components/text-placeholder';
 import useProductsQuery from 'calypso/a8c-for-agencies/data/marketplace/use-products-query';
-import { calculateCommissions } from '../lib/commissions';
-import type { Referral } from '../types';
+import { getConsolidatedData } from '../lib/commissions';
+import type { Referral, ReferralInvoice } from '../types';
+
+export default function CommissionsColumn( {
+	referral,
+	referralInvoices,
+}: {
+	referral: Referral;
+	referralInvoices: ReferralInvoice[];
+} ) {
+	const translate = useTranslate();
 
-export default function CommissionsColumn( { referral }: { referral: Referral } ) {
 	const { data, isFetching } = useProductsQuery( false, false, true );
 
-	const [ commissions, setCommissions ] = useState< number >( 0 );
+	const [ showPopover, setShowPopover ] = useState( false );
+	const wrapperRef = useRef< HTMLDivElement | null >( null );
+
+	const [ consolidatedData, setConsolidatedData ] = useState( {
+		allTimeCommissions: 0,
+		pendingOrders: 0,
+		pendingCommission: 0,
+	} );
 
 	useEffect( () => {
-		const commissions = calculateCommissions( referral, data || [] );
-		setCommissions( commissions );
-	}, [ referral, data ] );
+		if ( data?.length ) {
+			const consolidatedData = getConsolidatedData( [ referral ], data || [], referralInvoices );
+			setConsolidatedData( consolidatedData );
+		}
+	}, [ data, referral, referralInvoices ] );
+
+	const allTimeCommissions = formatCurrency( consolidatedData.allTimeCommissions, 'USD' );
+	const pendingCommission = formatCurrency( consolidatedData.pendingCommission, 'USD' );
 
-	return isFetching ? <TextPlaceholder /> : formatCurrency( commissions, 'USD' );
+	return isFetching ? (
+		<TextPlaceholder />
+	) : (
+		<>
+			<span
+				onMouseEnter={ () => setShowPopover( true ) }
+				onMouseLeave={ () => setShowPopover( false ) }
+				onMouseDown={ () => setShowPopover( false ) }
+				role="button"
+				tabIndex={ 0 }
+				ref={ wrapperRef }
+			>
+				{ allTimeCommissions }
+			</span>
+			<Tooltip context={ wrapperRef.current } isVisible={ showPopover } position="bottom">
+				<div>
```

Comment:
Do we need this extra div here for styling?

Reply:
No, we don't need this. Fixed it here: https://github.com/Automattic/wp-calypso/pull/92862/commits/d6c331cf89152b760a32a6446ba8ada88f5f7ca9